pkg/resource/scalable_target/custom_api.go

```diff
@@ -31,3 +32,19 @@ func (rm *resourceManager) customDescribeScalableTarget(
 		input.SetResourceIds(resourceIDList)
 	}
 }
+
+func (r *resource) customSetIdentifierCode(
+	identifier *ackv1alpha1.AWSIdentifiers,
+) {
+	r.ko.Spec.ResourceID = &identifier.NameOrID
+}
+
+// customCheckRequiredFieldsMissingMethod returns true if there are any fields
+// for the ReadOne Input shape that are required but not present in the
+// resource's Spec or Status
+func (rm *resourceManager) customCheckRequiredFieldsMissingMethod(
+	r *resource,
+) bool {
+	return r.ko.Spec.ResourceID == nil
```

Comment:
is service namespace also a required field?

Reply:
Technically, it is the only required field here - https://github.com/aws/aws-sdk-go/blob/main/models/apis/application-autoscaling/2016-02-06/api-2.json#L278